python-package/xgboost/compat.py

```diff
@@ -72,6 +73,22 @@ def lazy_isinstance(instance: Any, module: str, name: str) -> bool:
     XGBStratifiedKFold = None
 
 
+_logger = logging.getLogger(__name__)
+
+
+def is_cudf_available() -> bool:
+    """Check cuDF package available or not"""
+    if importlib.util.find_spec("cudf") is None:
+        return False
+    try:
+        import cudf
+
+        return True
+    except ImportError:
+        _logger.exception("Importing cuDF failed, use DMatrix instead of QDM")
```

Comment:
If an exception is raised, I don't think we can continue to use DMatrix instead of QDM right? Have you tested the code to verify that this statement holds true?

Reply:
uh ?

We can simply test:
```
>>> def f1():
...     try:
...             raise ValueError()
...             return True
...     except ValueError:
...             return False
... 
>>> 
>>> f1()
False
```

Reply:
I meant `        _logger.exception("Importing cuDF failed, use DMatrix instead of QDM")`, This will reraise the exception right?

Reply:
@trivialfis No. It only log message and exception stack, does not reraise exception.  See test:

```
>>> import logging
>>> 
>>> def f1():
...     try:
...         raise ValueError()
...         return True
...     except ValueError:
...         logging.exception("log exception.")
...         return False
... 
>>> print(f"f1 return value: {f1()}")
ERROR:root:log exception.
Traceback (most recent call last):
  File "<stdin>", line 3, in f1
ValueError
f1 return value: False
```
The `f1 return value: False` line is printed.

Reply:
Got it, thank you for the explanation.